.github/workflows/docs.yml

```diff
@@ -0,0 +1,29 @@
+name: docs
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ '*' ]
+
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-20.04
+    steps:
+    - name: Install Rust nightly
+      uses: actions-rs/toolchain@v1
+      with:
+          toolchain: nightly
+          override: true
+          components: rustfmt, clippy
+    - uses: actions/checkout@v3
+    - name: Make Docs
+      run: make docs
+    - name: Jekyll fix
+      run: touch target/doc/.nojekyll
+    - name: Update gh-pages
+      uses: JamesIves/github-pages-deploy-action@v4.3.3
```

Comment:
This action seems to attempt to force push. That's actually why the pipelines is failing right now. Maybe there's a way to configure it so it creates a PR instead?

Reply:
There is a `force` option enabled by default, with the stated alternative of 
> ..attempt to rebase new deployments onto any existing ones
when force is disabled. If that's not viable we may need another action entirely.